src/settings_internal.h

```diff
@@ -113,6 +113,8 @@ struct SettingDesc : SettingDescBase {
 	SettingType GetType() const;
 };
 
+typedef std::initializer_list<const SettingDesc> SettingTable;
```

Comment:
I tried googling about it, but nobody is giving an answer I can understand, but:

why not `std::array`? That seems to be the normal replacement for a `const T a[]` construct?

(I don't know enough about C++ to figure out the answer myself, so I am bugging you with it; sorry :D)